frontend/src/components/doctor/DoctorForm.tsx

```diff
@@ -10,36 +10,52 @@ import {
   Heading,
   Button,
 } from "@chakra-ui/react";
-import { useNavigate } from "react-router-dom";
 
-import { useState } from "react";
-
-import { DoctorDetail } from "./DetailDoctorPage";
+import { useEffect, useState } from "react";
+import { Doctor } from "./DoctorControl"; // type import
+import { useForm } from "react-hook-form";
 
 type Props = {
-  individualDoctorDetail?: DoctorDetail;
-  onSave: (details: DoctorDetail) => void;
+  individualDoctorDetail?: Doctor;
+  onSave: (details: Doctor) => void;
 };
 
-function DoctorForm({ individualDoctorDetail, onSave }: Props) {
-  const navigate = useNavigate();
-
-  const initialValue: DoctorDetail = {
-    id: Math.round(Math.random() * 1000),
-    first_name: "",
-    last_name: "",
-    email: "",
-    gender: "",
-    location: "",
-    insurance: [],
-    days_available: [],
-    speciality: "",
-    about: "",
-  };
-
-  let [inputValues, setInputValues] = useState(
-    individualDoctorDetail || initialValue
-  );
+const initialValue: Doctor = {
+  doctorId: Math.round(Math.random() * 1000), // need to work on these 3"
+  fkUserId: Math.round(Math.random() * 1000),
+  userId: Math.round(Math.random() * 1000),
+  firstName: "",
+  lastName: "",
+  email: "",
+  gender: "",
+  location: "",
+  status: true,
+  specialty: "",
+  about: "",
+};
+
+// const {
+//   register,
+//   handleSubmit,
+//   formState: { errors },
+// } = useForm();
+
+function DoctorForm({ individualDoctorDetail = initialValue, onSave }: Props) {
+  const [isActive, setIsActive] = useState(false);
+  // const navigate = useNavigate();
+
+  // ?? means if individualDoctorDetail is undefined or null use initialValue
+  if (individualDoctorDetail) {
+    console.log("using the value from parent component");
+  } else {
+    console.log("using the default values");
+  }
+  let [inputValues, setInputValues] = useState(individualDoctorDetail);
+
+  // force the state to be updated when the prop changes
+  useEffect(() => {
+    setInputValues(individualDoctorDetail);
+  }, [individualDoctorDetail]);
```

Comment:
This `useEffect` isn't necessary: https://beta.reactjs.org/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes